backend/src/test/java/com/github/lenaruppert/backend/service/JobServiceTest.java

```diff
@@ -138,4 +138,33 @@ void whenGetJobByIdWithNotValidId_thenThrowException() {
 
         verify(jobRepository).findById("2");
     }
+
+    @Test
+    void whenDeleteJobByIdWithExistingId_thenReturnClientToDelete() {
+        //GIVEN
+        when(jobRepository.findById(jobOne.id())).thenReturn(Optional.of(jobOne));
+        when(clientRepository.findById(jobOne.clientId())).thenReturn(Optional.of(clientOne));
+        //WHEN
+        Job actual = jobService.deleteJobById(clientOne.id());
```

Comment:
I think `clientOne.id()` is the wrong id. Shouldn't it be `jobOne.id()`? Probably the test works because they have the same id `1`, but logically you want to delete a job, not a client :)
You could also give the records in your tests ids like `j1` and `c1` to prevent these kind of errors.